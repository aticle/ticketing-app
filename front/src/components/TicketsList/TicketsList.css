.ag-theme-balham {
    min-width: 100%;
}