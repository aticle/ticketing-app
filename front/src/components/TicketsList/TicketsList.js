// @flow
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { AgGridReact } from 'ag-grid-react';

import type TicketState from '../Ticket/Ticket';
import type{ State } from '../../reducers';

import 'ag-grid-community/dist/styles/ag-grid.css';
import 'ag-grid-community/dist/styles/ag-theme-balham.css';
import './TicketsList.css';

type TicketsListState = {
    columnDefs: Array<Object>,
    rowData: Array<Object>
};

type TicketsListProps = {
    tickets: Array<TicketState>
};

class TicketsList extends Component<TicketsListProps, TicketsListState> {
    state = {
        columnDefs: [
            { headerName: "ID", field: "id" },
            { headerName: "Status", field: "status" },
            { headerName: "Title", field: "title" },
            { headerName: "Description", field: "description" }
        ],
        rowData: []
    };


    render() {
        return (
            <div
                style={{
                    height: '300px',
                    width: '500px'
                }}
                className="ag-theme-balham">
                <AgGridReact
                    columnDefs={this.state.columnDefs}
                    rowData={this.props.tickets}>
                </AgGridReact>
            </div>
        );
    }
}
export const mapStateToProps = (state: State) => {
    return {
        tickets: state.tickets
    };
};

export default connect(mapStateToProps)(TicketsList);
