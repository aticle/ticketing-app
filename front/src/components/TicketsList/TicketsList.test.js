
import React from 'react';
import { shallow } from 'enzyme';
import { AgGridReact } from 'ag-grid-react';

import TicketsList, { mapStateToProps } from './TicketsList';

describe('<TicketsList />', () => {
    it('renders correctly', () => {
        const ticketsList = shallow(<TicketsList.WrappedComponent />);
        expect(ticketsList).toMatchSnapshot();
    });

    it('renders an ag-grid', () => {
        const ticketsList = shallow(<TicketsList.WrappedComponent />);
        expect(ticketsList.find(AgGridReact)).toHaveLength(1);
    });

    it('mapStateToProps correctly', () => {
        expect(mapStateToProps({ tickets: [] })).toHaveProperty('tickets', []);
    });
});