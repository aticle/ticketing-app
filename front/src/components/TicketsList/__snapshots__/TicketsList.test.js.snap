// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TicketsList /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TicketsList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <AgGridReact
        columnDefs={
          Array [
            Object {
              "field": "id",
              "headerName": "ID",
            },
            Object {
              "field": "status",
              "headerName": "Status",
            },
            Object {
              "field": "title",
              "headerName": "Title",
            },
            Object {
              "field": "description",
              "headerName": "Description",
            },
          ]
        }
      />,
      "className": "ag-theme-balham",
      "style": Object {
        "height": "300px",
        "width": "500px",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "columnDefs": Array [
          Object {
            "field": "id",
            "headerName": "ID",
          },
          Object {
            "field": "status",
            "headerName": "Status",
          },
          Object {
            "field": "title",
            "headerName": "Title",
          },
          Object {
            "field": "description",
            "headerName": "Description",
          },
        ],
        "rowData": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <AgGridReact
          columnDefs={
            Array [
              Object {
                "field": "id",
                "headerName": "ID",
              },
              Object {
                "field": "status",
                "headerName": "Status",
              },
              Object {
                "field": "title",
                "headerName": "Title",
              },
              Object {
                "field": "description",
                "headerName": "Description",
              },
            ]
          }
        />,
        "className": "ag-theme-balham",
        "style": Object {
          "height": "300px",
          "width": "500px",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "columnDefs": Array [
            Object {
              "field": "id",
              "headerName": "ID",
            },
            Object {
              "field": "status",
              "headerName": "Status",
            },
            Object {
              "field": "title",
              "headerName": "Title",
            },
            Object {
              "field": "description",
              "headerName": "Description",
            },
          ],
          "rowData": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenThreeAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
